p2p/starknet/client.go

```diff
@@ -0,0 +1,127 @@
+package starknet
+
+import (
+	"context"
+
+	"github.com/NethermindEth/juno/p2p/starknet/spec"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"google.golang.org/protobuf/proto"
+)
+
+type NewStreamFunc func(ctx context.Context, pids ...protocol.ID) (network.Stream, error)
```

Comment:
very pedantic nit: this is part of the `libp2p/host.Host` interface. Can we make `Streamer` interface with this function so we can pass the host directly to `NewClient`?

Reply:
I like function types better since an interface would make testing a bit harder. With an interface we will have to create mock types instead of just a lambda.